pkg/cloudprovider/provider/gce/provider.go

```diff
@@ -265,7 +265,9 @@ func (p *Provider) Create(machine *clusterv1alpha1.Machine, data *cloudprovidert
 		Disks:             disks,
 		Labels:            labels,
 		Scheduling: &compute.Scheduling{
-			Preemptible: cfg.preemptible,
+			Preemptible:       cfg.preemptible,
+			AutomaticRestart:  &cfg.automaticRestart,
```

Comment:
This is an incomplete implementation as you are not consuming values from `ProviderSpec` for the new fields i.e. AutomaticRestart and ProvisioningModel.

Please go through the handling of `preemptible` in `newConfig` as an example:

https://github.com/kubermatic/machine-controller/blob/master/pkg/cloudprovider/provider/gce/config.go#L165

Reply:
You are totally right, sorry for that!